cfme/infrastructure/host.py

```diff
@@ -497,111 +422,93 @@ def untag(self, tag):
         mixins.remove_tag(tag)
 
 
-@navigator.register(Host, 'All')
+@navigator.register(Host)
 class All(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+    VIEW = HostsView
+    prerequisite = NavigateToAttribute("appliance.server", "LoggedIn")
 
     def step(self):
         try:
-            self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
+            self.prerequisite_view.navigation.select("Compute", "Infrastructure", "Hosts")
         except NoSuchElementException:
-            self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Nodes')
+            self.prerequisite_view.navigation.select("Compute", "Infrastructure", "Nodes")
 
     def resetter(self):
-        tb.select("Grid View")
-        sel.check(paginator.check_all())
-        sel.uncheck(paginator.check_all())
+        if self.view.toolbar.view_selector.selected != "Grid View":
+            self.view.toolbar.view_selector.select("Grid View")
+        self.view.paginator.check_all()
```

Comment:
and I'm not sure we need to switch view to Grid now. Since entity object doesn't depend on currently chosen view now.